zkevm-circuits/src/bytecode_circuit/bytecode_unroller.rs

```diff
@@ -715,6 +716,7 @@ mod tests {
 
     /// Tests a circuit with incomplete bytecode
     #[test]
+    #[should_panic = "called `Result::unwrap()` on an `Err` value: NotEnoughRowsAvailable { current_k: 9 }"]
```

Comment:
If it's incomplete how can it panic due to `NotEnoughRowsAvailable`??

Reply:
AFAIU `incomplete` here means that not all code can be processed, but probally there's another definition that I'm missing.

Reply:
@Brechtpd, I see, thanks!

Yes, I can make the circuit failing as you said by not adding rows when assigning witness if they are going to overflow the table, and works,

```
341                        if offset <= last_row_offset {
342                             // Set the data for this row
343                             self.set_row(
```

Nonetheless, I'm asking myself if creating witness that makes a circuit failing a good strategy for tests. Is not just better to fail with `NotEnoughRowsAvailable`? 

Anyway we are going to change how bytecode circuits works, so maybe we do not have to spend more time on this issue :)

Reply:
> Nonetheless, I'm asking myself if creating witness that makes a circuit failing a good strategy for tests. Is not just better to fail with NotEnoughRowsAvailable?

The main goal of the circuits is to reject incorrect data, and so the only way to really test that is to check if the circuit indeed cannot be verified with that data. The fact that halo2 library throws an error assigning the witness is actually not useful from a security pov because that is something a malicious prover can just work around by changing a bit of code when he generates the proof.

> Anyway we are going to change how bytecode circuits works, so maybe we do not have to spend more time on this issue :)

Yeah true :), but I think the tests can remain the same because the change is just how the circuit does some of the constraints, and so the tests still need to test the same things.

Reply:
> The main goal of the circuits is to reject incorrect data, and so the only way to really test that is to check if the circuit indeed cannot be verified with that data. The fact that halo2 library throws an error assigning the witness is actually not useful from a security pov because that is something a malicious prover can just work around by changing a bit of code when he generates the proof.

Super good information to keep in mind! Thanks!

> Yeah true :), but I think the tests can remain the same because the change is just how the circuit does some of the constraints, and so the tests still need to test the same things.

Fixed in https://github.com/appliedzkp/zkevm-circuits/pull/403/commits/f33f782b5960233e13518f96a90e99e696b4651f (I hope!)